lib/rubocop/cop/gemspec/required_ruby_version.rb

```diff
@@ -0,0 +1,64 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Gemspec
+      # Checks that `require_ruby_version` of gemspec and `TargetRubyVersion`
+      # of .rubocop.yml are equal.
+      # Thereby, RuboCop to perform static analysis working on the version
+      # required by gemspec.
+      #
+      # @example
+      #   # When `TargetRubyVersion` of .rubocop.yml is `2.3`.
+      #
+      #   # bad
+      #   Gem::Specification.new do |spec|
+      #     spec.required_ruby_version = '>= 2.2.0'
+      #   end
+      #
+      #   # bad
+      #   Gem::Specification.new do |spec|
+      #     spec.required_ruby_version = '>= 2.4.0'
+      #   end
+      #
+      #   # good
+      #   Gem::Specification.new do |spec|
+      #     spec.required_ruby_version = '>= 2.3.0'
+      #   end
+      #
+      #   # good
+      #   Gem::Specification.new do |spec|
+      #     spec.required_ruby_version = '>= 2.3'
+      #   end
+      #
+      #   # good
+      #   Gem::Specification.new do |spec|
+      #     spec.required_ruby_version = ['>= 2.3.0', '< 2.5.0']
+      #   end
+      class RequiredRubyVersion < Cop
+        MSG = '`require_ruby_version` and `TargetRubyVersion` of ' \
```

Comment:
I'd mention in the message the actual versions that were detected. This would make it easier to understand this message.

Reply:
Certainly that information is useful, which makes it easier for users to take action against offense. I updated this message by 0d33957. Since that change had warned by `Metrics/AbcSize` cop, I refactored by Extract Method in this commit.

Reply:
Now the message is too informative I think. Let's shorten it to the following:
``require_ruby_version` (x.x, declared in x.gemspect) and `TargetRubyVersion` (x.x, declared in .rubocop.yml) should be equal.`

Reply:
Thanks for suggesting the message. I updated by c735ed0.